### PR TITLE
Added show angles option to line and area tools

### PR DESCRIPTION
**Pull request by int-Frank**
Fixes [AB#1837](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/1837)

### CODE REVIEW

**Comment by jproust** [comment]
Angles_issue
What side is the 177.27 deg angle

**Comment by int-Frank** [comment]
> Angles_issue
> What side is the 177.22 deg angle

I use the  _smallest_ angle between the lines. Or in other words, the inverse cosine of the dot product between the vectors of the two lines.

**Comment by jproust** [comment]
> I use the _smallest_ angle between the lines. Or in other words, the inverse cosine of the dot product between the vectors of the two lines.

I know I have read your code, I just find the display of the angle a bit confusing when it's close to 180° as the user can't really know which side the angle is. It can be confusing if for instance the smallest angle is an angle in the z plane and the user is watching from above.
Anyway we can display the measured angle ?
angle_issue

File: src/scene/vcPOI.cpp [hunk 1]
```diff
@@ -68,6 +68,9 @@ class vcPOIState_General
       if (ImGui::Checkbox(udTempStr("%s##POIShowAllLengths%zu", vcString::Get("scenePOILineShowAllLengths"), itemID), &m_pParent->m_showAllLengths))
         udProjectNode_SetMetadataBool(m_pParent->m_pNode, "showAllLengths", m_pParent->m_showAllLengths);
 
+      if (ImGui::Checkbox(udTempStr("%s##POIShowAngles%zu", vcString::Get("scenePOILineShowAngles"), itemID), &m_pParent->m_showAngles))
+        udProjectNode_SetMetadataBool(m_pParent->m_pNode, "showAngles", m_pParent->m_showAngles);
```
Comment (OsiJr) [inline src/scene/vcPOI.cpp:72]: `OnNodeUpdate` doesn't load this key (there is a helper in vcProject for bools)

**Review by OsiJr — CHANGES_REQUESTED** [review]
Will need to load the show angles setting

**Comment by int-Frank** [comment]
@jproust Ah I see. Perhaps I could render a little angle symbol between the lines. Something like 
image
 @OsiJr What do you think?

**Review by OsiJr — APPROVED** [review]

**Review by OsiJr — APPROVED** [review]

**Pull request merged**